packages/services/src/state_committer/eigen_service.rs

```diff
@@ -0,0 +1,191 @@
+use metrics::{
+    RegistersMetrics,
+    prometheus::{IntGauge, Opts, core::Collector},
+};
+use tracing::info;
+
+use crate::{
+    Result, Runner,
+    types::{AsB64, storage::BundleFragment},
+};
+
+use super::commit_helpers::update_current_height_to_commit_metric;
+
+// src/config.rs
+#[derive(Debug, Clone)]
+pub struct Config {
+    // The throughput of the da layer API in MB/s.
+    pub api_throughput: u32,
+    /// The lookback window in blocks to determine the starting height.
+    pub lookback_window: u32,
+}
+
+#[cfg(feature = "test-helpers")]
+impl Default for Config {
+    fn default() -> Self {
+        Self {
+            api_throughput: 16,
+            lookback_window: 1000,
+        }
+    }
+}
+
+struct Metrics {
+    current_height_to_commit: IntGauge,
+}
+
+impl Default for Metrics {
+    fn default() -> Self {
+        let current_height_to_commit = IntGauge::with_opts(Opts::new(
+            "current_height_to_commit",
+            "The starting l2 height of the bundle we're committing/will commit next",
+        ))
+        .expect("metric config to be correct");
+
+        Self {
+            current_height_to_commit,
+        }
+    }
+}
+
+impl<DALayer, FuelApi, Db, Clock> RegistersMetrics for StateCommitter<DALayer, FuelApi, Db, Clock> {
+    fn metrics(&self) -> Vec<Box<dyn Collector>> {
+        vec![Box::new(self.metrics.current_height_to_commit.clone())]
+    }
+}
+
+/// The `StateCommitter` is responsible for committing state fragments to L1.
+pub struct StateCommitter<DALayer, FuelApi, Db, Clock> {
+    da_layer: DALayer,
+    fuel_api: FuelApi,
+    storage: Db,
+    config: Config,
+    clock: Clock,
+    metrics: Metrics,
+}
+
+impl<DALayer, FuelApi, Db, Clock> StateCommitter<DALayer, FuelApi, Db, Clock>
+where
+    Clock: crate::state_committer::port::Clock,
+{
+    /// Creates a new `StateCommitter`.
+    pub fn new(
+        da_layer: DALayer,
+        fuel_api: FuelApi,
+        storage: Db,
+        config: Config,
+        clock: Clock,
+    ) -> Self {
+        Self {
+            da_layer,
+            fuel_api,
+            storage,
+            config,
+            clock,
+            metrics: Metrics::default(),
+        }
+    }
+}
+
+impl<DALayer, FuelApi, Db, Clock> StateCommitter<DALayer, FuelApi, Db, Clock>
+where
+    DALayer: crate::state_committer::port::eigen_da::Api + Send + Sync,
+    FuelApi: crate::state_committer::port::fuel::Api,
+    Db: crate::state_committer::port::Storage,
+    Clock: crate::state_committer::port::Clock,
+{
+    async fn submit_fragment(&self, fragment: BundleFragment) -> Result<()> {
```

Comment:
do we want to add some retry mechanism here (exponential backoff or something)

Reply:
all subservices in the block committer run on an interval, so this is implicitly handled by the scheduling of the task :)